### PR TITLE
Refactoring: shared utilities and cleanup

### PR DESCRIPTION
**Pull request by dandavison**
This is pure refactoring aimed at slightly reducing friction when adding new features:

- New harness utilities for starting a workflow with default options: `start_parameterless_workflow` and `executeParameterlessWorkflow` for Python and Typescript respectively.

- Convert `child_workflow/result` Java feature to use single `feature.java` file (and move `ChildWorkflow` impls inside `ChildWorkflow` interface)

- Make use of `runner.executeSingleParameterlessWorkflow` in a Java feature that wasn't using it.

### CODE REVIEW

File: features/child_workflow/signal/feature.java [hunk 1]
```diff
@@ -13,37 +11,36 @@
...
+        return childWorkflowUnblockMessage;
+      }
 
-    @Override
-    public void unblock(String message) {
-      childWorkflowUnblockMessage = message;
+      @Override
+      public void unblock(String message) {
+        childWorkflowUnblockMessage = message;
+      }
```
Comment (dandavison) [inline features/child_workflow/signal/feature.java:43]: The diff here is mostly indentation, moving an impl into the interface - easier to understand with hide-whitespace on.

File: features/child_workflow/result/feature.java [hunk 2]
```diff
@@ -5,52 +5,66 @@
 import io.temporal.sdkfeatures.Feature;
```
Comment (dandavison) [inline features/child_workflow/result/feature.java:5]: The diff in the file is easier to understand with hide-whitespace on: spotless converting 4-space to 2-space indentation.

**Comment by dandavison** [comment]
I briefly looked into changing the Java harness so that we can define multiple workflows symmetrically (rather than having one standalone workflow interface and the other workflow interface implemented by the feature), but it's fairly deeply engrained in the harness design so I left that.

File: features/child_workflow/result/feature.java (hunk 2 above)
Comment (cretz) [reply features/child_workflow/result/feature.java:5]: Greatly appreciated! I think on the Java SDK there is an auto-formatter. I wonder if we want to bring that over here.

**Review by cretz — APPROVED** [review]

**Comment by dandavison** [comment]
Thanks for reviewing @cretz!

**Pull request merged**